codex/proving/simulated.nim

```diff
@@ -0,0 +1,54 @@
+import ../conf
+when codex_enable_proof_failures:
+
+  import std/strutils
+  import pkg/chronicles
+  import pkg/ethers
+  import pkg/ethers/testing
+  import ../market
+  import ../clock
+  import ./proving
+
+  type
+    SimulatedProving* = ref object of Proving
+      failEveryNProofs: uint
+      proofCount: uint
+
+  logScope:
+    topics = "simulated proving"
+
+  func new*(_: type SimulatedProving,
+            market: Market,
+            clock: Clock,
+            failEveryNProofs: uint): SimulatedProving =
+
+    let p = SimulatedProving.new(market, clock)
+    p.failEveryNProofs = failEveryNProofs
+    return p
+
+  method init(proving: SimulatedProving) {.async.} =
+    if proving.failEveryNProofs > 0'u and await proving.market.isMainnet():
```

Comment:
I am not sure about this. IMHO I would remove the `isMainnet` check. First of all, we don't know if we gonna even run on Mainnet (most probably not). I guess you are trying to be on the safe side in case somebody finds binary compiled with the simulated-proving enabled and tried to run it on Mainnet, right? IMHO it is a low probability as they would also have to specify the config parameter to enable this and if they would really want to run this on mainnet then they can always compile their own version to work around this restriction...

Reply:
Yes, this is mostly an artifact leftover from before adding the compilation flag. Will remove.